Feature/219 accs delete setting

ceiroa opened the pull request:
# Warning

# Info
- Multi-addresses now supported in any Account record type existing in the org.
- Admin can how choose with type of Accounts will be automatically deleted if all its children Contacts have been deleted.

# Issues
Fixes #219
Fixes #220

ceiroa commented:
Hi @kbromer ~ There are a lot of changes that went into this pull request, including a new reusable lightning component, and some Javascript refactoring. Do you think you could review it for me? Thanks!

davidhabib commented on src/aura/STG_CMP_Base/STG_CMP_BaseHelper.js:
```diff
@@ -81,6 +56,14 @@
 		}
 	},
 	
+	addPrefix : function(namespacePrefix, key) {
```
I assume the namespace prefix contains the dot in all this code?

ceiroa replied:
Actually, we are manipulating custom settings fields, and the namespace needs to have the double underscore. This is done in https://github.com/SalesforceFoundation/HEDAP/blob/master/src/pages/STG_Settings.page#L29.

davidhabib commented on src/aura/STG_CMP_System/STG_CMP_System.cmp:
```diff
@@ -14,10 +13,10 @@
-                        <ui:inputSelectOption label="{!recType.devName}" text="{!recType.id}" 
-                            value="{!v.accRecTypeName == recType.devName ? true : false}" />
+                        <ui:inputSelectOption label="{!recType.name}" text="{!recType.id}" 
+                            value="{!v.accRecTypeId == recType.id ? true : false}" />
```
why the change from using recordtype Developer names to using recordtype Id's?

ceiroa replied:
Just to standardize storing recordtype IDs in all settings. The change to using dev names was done recently (https://github.com/SalesforceFoundation/HEDAP/pull/226/), as an intermediate step.

davidhabib commented on src/package.xml:
```diff
@@ -214,15 +215,15 @@
         <members>Hierarchy_Settings__c.Enable_Debug__c</members>
         <members>Hierarchy_Settings__c.Error_Notifications_On__c</members>
         <members>Hierarchy_Settings__c.Error_Notifications_To__c</members>
-        <members>Hierarchy_Settings__c.Household_Adm_Acct_Addresses_Enabled__c</members>
```
can you delete these if they were already packaged?

ceiroa replied:
I think you can delete custom settings fields the same you can delete object fields.

ceiroa commented:
@davidhabib ~ This is ready for another look. I moved those checkboxes we talked about to the right, and since I was on it, I implemented the same pattern for all checkboxes across settings. The others didn't have labels, but this make them all have the slds look also in edit mode.
